odoo/tools/date_utils.py

```diff
@@ -163,9 +165,14 @@ def parse_date(value: str, env: Environment) -> date | datetime:
             raise ValueError(f"Invalid term {term!r} in expression date: {value!r}")
 
         # Weekday
-        if (weekday := WEEKDAY_NUMBER.get(term[1:])) is not None:
+        if term[1:] == "week_start":
+            weekday = week_start
```

Comment:
Only execute this when needed.
```suggestion
            weekday = int(env["res.lang"]._get_data(code=env.user.lang).week_start) - 1
```

Reply:
I also need it for the offset calculations

Reply:
@kmagusiak I moved the initialization of `week_start` to inside the loop. Is that what you meant?